coreax/util.py

```diff
@@ -239,7 +246,10 @@ def solve_qp(kernel_mm: ArrayLike, gramian_row_mean: ArrayLike, **osqp_kwargs) -
     sol = qp.run(
         params_obj=(q_array, c), params_eq=(a_array, b), params_ineq=(g_array, h)
     ).params
-    return sol.primal
+
+    # Ensure conditions of solution are met to chosen precision
+    solution = jnp.where(sol.primal < jnp.abs(precision_threshold), 0, sol.primal)
```

Comment:
I think it might be worth adding a unittest to verify the threshold is applied correctly. Something like, mock/patch the qp.run, forcing it to return some values above the precision threshold and some values below, then verify that the ones below have been filled in.

Reply:
I forgot that we already have a tested `apply_negative_precision_threshold` function so I have now used that instead